### PR TITLE
feat: support ROS 2 Humble

### LINKED ISSUE #329: Support ROS 2 Humble
### Checklist

- [X] I've read the [contribution guidelines](https://github.com/autowarefoundation/autoware/blob/main/CONTRIBUTING.md).
- [X] I've searched other issues and no duplicate issues were found.
- [X] I've agreed with the maintainers that I can plan this task.

### Description

A follow-up for #238.

### Purpose

To support ROS 2 Humble.

### Possible approaches

Improve #239.

### Definition of done

Autoware's Humble Docker image is published.

### PR DESCRIPTION
**Pull request by kenji-miyake**
## Description

Resolves autowarefoundation/autoware.ai#2443.

I won't merge this PR to `main` until `Galactic` support by the AWF ends.
Users have to switch branches if they want to use `Humble` because it would be too complex if we support both distros in one branch.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by ghost** [comment]
Given that it may take Autoware users some time to upgrade all of their dev platforms and ECUs to Ubuntu 22.04, would it perhaps be better to merge this PR into main in advance of Galactic reaching end of support? 🤔

**Comment by kenji-miyake** [comment]
@LalithVipulananthan Yes, the current proposal is around the end of August.
https://github.com/autowarefoundation/autoware/discussions/358

File: autoware.repos [hunk 1]
```diff
@@ -86,3 +86,21 @@ repositories:
...
+    version: rolling
+  missing_packages/pacmod3_msgs:
+    type: git
+    url: https://github.com/astuff/pacmod3_msgs.git
+    version: main
+  # https://github.com/gazebo-release/gz-math6-release/issues/11
+  missing_packages/velodyne_simulator:
+    type: git
+    url: https://github.com/tier4/velodyne_simulator.git
+    version: prepare/humble
```
Comment (wep21) [inline autoware.repos]:
```suggestion
```

Comment (wep21) [reply autoware.repos]: @kenji-miyake I think the issue has been resolved.

**Comment by kenji-miyake** [comment]
Build CI:
- https://github.com/autowarefoundation/autoware/runs/6577553569?check_suite_focus=true
- https://github.com/autowarefoundation/autoware/runs/6577553103?check_suite_focus=true

File: autoware.repos (hunk 1 above)
Comment (kenji-miyake) [reply autoware.repos]: Failed with arm64. :thinking: 
https://github.com/autowarefoundation/autoware/runs/6573429005?check_suite_focus=true#step:6:1959

Comment (kenji-miyake) [reply autoware.repos]: Asked at https://answers.ros.org/question/401274/why-velodyne_description-isnt-released-for-arm64/.
-> https://github.com/ros2-gbp/velodyne_simulator-release/issues/1

**Comment by kenji-miyake** [comment]
Similar problem to https://github.com/gazebo-release/gz-math6-release/issues/11.
https://github.com/autowarefoundation/autoware/runs/6585085834?check_suite_focus=true

```
 libjson-c-dev : Depends: libjson-c5 (= 0.15-2build4) but 0.15-3~ubuntu1.22.04.1 is to be installed
E: Unable to correct problems, you have held broken packages.
ERROR: the following rosdeps failed to install
  apt: command [apt-get install -y libpcl-dev] failed
```

But `rosdep install` in the build CI passed? :thinking: 
https://github.com/autowarefoundation/autoware/runs/6585145537?check_suite_focus=true

**Comment by wep21** [comment]
@kenji-miyake There are some conflicts. Could you resolve them?

**Comment by kenji-miyake** [comment]
A new error? :thinking: 
https://github.com/autowarefoundation/autoware/runs/7015153176?check_suite_focus=true#step:8:7434

```
The following packages have unmet dependencies:
 libudev-dev : Depends: libudev1 (= 249.11-0ubuntu3) but 249.11-0ubuntu3.3 is to be installed
E: Unable to correct problems, you have held broken packages.
ERROR: the following rosdeps failed to install
  apt: command [apt-get install -y ros-humble-joy] failed
```

**Comment by kenji-miyake** [comment]
Moved to https://github.com/autowarefoundation/autoware/pull/2692.

**Pull request closed without merging**